CMakeLists.txt

```diff
@@ -2023,6 +2024,52 @@ if((CMAKE_BUILD_TYPE IN_LIST build_types) AND (NOT NO_BUILD_TYPE_WARNING))
   endif()
 endif()
 
+if(CONFIG_LLEXT_EDK)
+    set(LLEXT_EDK_FILE ${PROJECT_BINARY_DIR}/${CONFIG_LLEXT_EDK_NAME}.tar.xz)
+
+    # TODO maybe generate flags for C CXX ASM
+    zephyr_get_compile_options_for_lang(C zephyr_flags)
+
+    # Filter out non LLEXT and LLEXT_EDK flags - and add required ones
+    llext_filter_zephyr_flags(LLEXT_REMOVE_FLAGS ${zephyr_flags} llext_cflags)
+    llext_filter_zephyr_flags(LLEXT_EDK_REMOVE_FLAGS ${llext_cflags} llext_cflags)
+
+    list(APPEND llext_cflags ${LLEXT_APPEND_FLAGS})
+    list(APPEND llext_cflags ${LLEXT_EDK_APPEND_FLAGS})
+
+    add_custom_command(
+        OUTPUT ${LLEXT_EDK_FILE}
+        # Regenerate syscalls in case CONFIG_LLEXT_EDK_USERSPACE_ONLY
+        COMMAND ${CMAKE_COMMAND}
+            -E make_directory edk/include/generated
+        COMMAND
+            ${PYTHON_EXECUTABLE}
+            ${ZEPHYR_BASE}/scripts/build/gen_syscalls.py
```

Comment:
Can't this share all the boilerplate for how to call the python script with the userspace layer?  Or better, share the targets.  It's not clear to me if an llext SDK generation ever happens in the same build as a USERSPACE=y app build, but in theory they might and if they do, we don't want to  be building everything twice.

Reply:
I kinda expect the edk to be generated much less frequently than the app - even on CI, I would expected the edk to be generated only after the app passes all its own tests - but can't really assume that. 
That said, things are only rebuilt "needlessly" here if `CONFIG_LLEXT_EDK_USERSPACE_ONLY` is not set, as in this case we could just reuse the files built for the app. Otherwise, we need to rebuild them to skip the "inside kernel check".
I'd say that we can tackle this if this sometimes extra build becomes too burdensome - thoughts?

Reply:
Not sure I fully follow.

Is it a problem for the EDK if the syscall stubs contains routing code ?
Ref: 
> If the EDK is expected to be used by userspace only extensions, this option will make EDK stubs not contain the routing code, and only generate the userspace one.

The edk has a dependency to the Zephyr target here: 
https://github.com/zephyrproject-rtos/zephyr/blob/18090582613238a2bbee1e4c9493550453db104b/CMakeLists.txt#L2067

which means that Zephyr and thus the `gen_syscalls.py` will always be brought up-to-date before this code executes.
So exactly when can't the already generated syscall header and dispatch table just be reused ?

Does it provide a problem if the code excluded by the `gen_syscalls.py --userspace-only` flag is present ?